File: Numerical_Methods_for_Physics_Garcia/edition2/Fortran/ifft.f
      subroutine ifft( A, N )
      integer*4 N
      complex*16 A(*)
! Routine to compute inverse Fourier transform using FFT algorithm
! Inputs
!    A         Complex transform data vector
!    N         Number of data points
! Outputs
!    A         Complex time series

      integer*4 i

      !* Take complex conjugate of input transform
      do i=1,N
        A(i) = conjg(A(i))     ! Complex conjugate
      enddo

      !* Evaluate fast fourier transform
      call fft( A, N )

      !* Take complex conjugate and normalize by N
      do i=1,N
        A(i) = conjg(A(i))/N  ! Normalize and complex conjugate
      enddo

      return
      end

File: Numerical_Methods_for_Physics_Garcia/edition2/Cpp/ifft.cpp
#include "NumMeth.h"

void fft( Matrix& RealA, Matrix& ImagA);

void ifft( Matrix& RealA, Matrix& ImagA) {
// Routine to compute inverse Fourier transform using FFT algorithm
// Inputs
//    RealA, ImagA         Real and imaginary parts of transform
// Outputs
//    RealA, ImagA         Real and imaginary parts of time series

  int i, N = RealA.nRow();   // Number of data points

  //* Take complex conjugate of input transform
  for( i=1; i<=N; i++ )
    ImagA(i) *= -1.0;        // Complex conjugate

  //* Evaluate fast fourier transform  
  fft( RealA, ImagA );       
  
  //* Take complex conjugate and normalize by N
  double invN = 1.0/N;
  for( i=1; i<=N; i++ ) {
    RealA(i) *= invN;
    ImagA(i) *= -invN;    // Normalize and complex conjugate
  }
}
